src/v/datalake/record_schema_resolver.cc

```diff
@@ -60,9 +64,46 @@ struct schema_translating_visitor {
         }
     }
     ss::future<checked<type_and_buf, record_schema_resolver::errc>>
-    operator()(const ppsr::protobuf_schema_definition&) {
-        // XXX: a subsequent PR will add protobuf support.
-        co_return type_and_buf::make_raw_binary(std::move(buf_no_id));
+    operator()(const ppsr::protobuf_schema_definition& pb_def) {
+        const google::protobuf::Descriptor* d;
+        proto_offsets_message_data offsets;
+        try {
+            auto offsets_res = get_proto_offsets(buf_no_id);
+            if (offsets_res.has_error()) {
+                co_return record_schema_resolver::errc::bad_input;
+            }
+            offsets = std::move(offsets_res.value());
+            // TODO: maybe there's another caching opportunity here.
```

Comment:
q (curious):  you mean caching the `google::protobuf::Descriptor` itself yeah? offhand do you have an idea of what the key would look like? would you need to add more metadata to the visitor, or is the ID and the offsets likely enough?

Reply:
Right, exactly. I was thinking the just schema id + pb offsets (basically the datalake::schema_identifier), especially since this is going to be resolved per record, and across multiple partitions for the same topic.

Though maybe this parsing will be negligible if the common case is an empty list of offsets.